analyzers/src/SonarAnalyzer.CSharp.Styling/Rules/LambdaParameterName.cs

```diff
@@ -0,0 +1,38 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2024 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+namespace SonarAnalyzer.Rules.CSharp.Styling;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp)]
+public sealed class LambdaParameterName : StylingAnalyzer
+{
+    public LambdaParameterName() : base("T0010", "Use 'x' for the lambda parameter name.") { }
+
+    protected override void Initialize(SonarAnalysisContext context) =>
+        context.RegisterNodeAction(c =>
+            {
+                var parameter = ((SimpleLambdaExpressionSyntax)c.Node).Parameter;
+                if (parameter.Identifier.ValueText != "x" && c.Node.Parent.FirstAncestorOrSelf<LambdaExpressionSyntax>() is null)
```

Comment:
One question about this:
In the coding style, we say we will exclude[ roslyn callbacks](https://github.com/SonarSource/sonar-dotnet/blob/master/docs/coding-style.md#naming-conventions:~:text=for%20context%20of-,Roslyn%20callback.,-Short%20names%20can) as we usually name them `c`.
Should we do a semantic check to exclude all the callbacks that expect one of the roslyn callbacks?

Reply:
I just realized that's what Sebastien meant with `This lambda` on the comment above 🤦‍♂️

Reply:
Yes, Seb pointed that out in his last comment. It's done in the review update commit.